### PR TITLE
Slack plugin support

### PR DESCRIPTION
**Pull request by ibelikov**

### CODE REVIEW

File: modules/slack/resources/schema.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+title: Schema for Jenkins Slack Notification plugin configuration
+type: object
+additionalProperties: false
+
+properties:
+  team_subdomain:
```
Comment (akaRem) [inline modules/slack/resources/schema.yaml]: I'd propose to replace this with just slack url. There could be on-premise installations in future. And it's just easier to copypaste url than try to understand, which part of it is needed

File: modules/slack/resources/schema.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    type: string
+    description: |
+      Your team's Slack subdomain. If you sign in to slack at https://example.slack.com/, your subdomain is 'example'.
+  token:
+    type: string
+    description: |
+      The integration token to be used to send notifications to Slack.
+      You can copy this from the settings page within Slack.
+  token_credentials_id:
```
Comment (akaRem) [inline modules/slack/resources/schema.yaml]: why "credentials"? it's just only one.

File: modules/slack/tests/test_slack.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        self.mfs.add_entries({os.path.join(jimmy_dir, 'lib', 'schema.yaml'): self.jimmy_schema,
+                              os.path.join(jimmy_dir, 'jimmy.yaml'): self.mock_jimmy_yaml,
+                              slack_schema_path: mock_slack_schema,
+                              jenkins_yaml_path: '\n'.join(
+                                  [
+                                      'jenkins:',
+                                      '  slack:',
+                                      '    team_subdomain: slackteam',
+                                      '    token: access-token',
+                                      '    channel: build-notifications',
```
Comment (akaRem) [inline modules/slack/tests/test_slack.py]: "#build-notifications"

File: sample/input/jenkins.yaml [hunk 4]
```diff
@@ -113,6 +113,13 @@ jenkins:
       max_total_concurrent_builds: 1
       max_concurrent_builds_per_node: 0
 
+  slack:
+    team_domain: slackteam
+    token: slack-access-token
+    channel: build-notifications
```
Comment (akaRem) [inline sample/input/jenkins.yaml]: "#build-notifications"

File: modules/slack/resources/schema.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    description: |
+      Enter the channel names to which notifications should be sent. Note that this can include
+      names of channels OR channel id numbers, e.g. "#builds", and that multiple values may appear comma separated.
+      While names are more readable, channel ids will not change over time and are therefore more resilient.
+  webhook_token:
+    type: string
+    description: |
+      This token is used to verify requests between Slack and Jenkins.
+      You can copy this token from the settings page for your outgoing webhook within Slack.
+  webhook_endpoint:
```
Comment (akaRem) [inline modules/slack/resources/schema.yaml]: I'd propose to use webhook_url here. "Url" is much transparent than "endpoint"

File: modules/slack/resources/schema.yaml (hunk 1 above)
Comment (ibelikov) [inline modules/slack/resources/schema.yaml]: > There could be on-premise installations in future.

Premature optimisation fan?)

Feel free to fix this upstream https://github.com/jenkinsci/slack-plugin, I'm strongly opposed to adding some parsing logic here for this.

File: modules/slack/resources/schema.yaml (hunk 2 above)
Comment (ibelikov) [inline modules/slack/resources/schema.yaml]: Well, I was about to write something about 'credentials' as the commonly used mostly plural form in this sense, but it is 'credential' in upstream plugin, so you're probably right here.

File: modules/slack/resources/schema.yaml (hunk 5 above)
Comment (ibelikov) [inline modules/slack/resources/schema.yaml]: ok

File: modules/slack/tests/test_slack.py (hunk 3 above)
Comment (ibelikov) [inline modules/slack/tests/test_slack.py]: forgot about direct messages, yep.

File: modules/slack/resources/schema.yaml (hunk 1 above)
Comment (akaRem) [inline modules/slack/resources/schema.yaml]: Ok then. I didn't look at plugin's readme.

**Pull request merged**